packages/core/src/lib/utils/url.ts

```diff
@@ -0,0 +1,14 @@
+export interface ConnectionConfig {
+  ssl?: boolean | undefined;
+  host?: string | undefined;
```

Comment:
You have added the `?`, is means undefined, so actually, you don't need to write again by `undefined`.